src/batch.jl

```diff
@@ -0,0 +1,199 @@
+"""
+    BatchIntegrand(f!, y::AbstractVector, x::AbstractVector, max_batch=typemax(Int))
+
+Constructor for a `BatchIntegrand` accepting an integrand of the form `f!(y,x) = y .= f.(x)`
+that can evaluate the integrand at multiple quadrature nodes using, for example, threads,
+the GPU, or distributed-memory. The `max_batch` keyword roughly limits the number of nodes
+passed to the integrand, though at least `4*order+2` nodes will be used by the GK rule.
+The buffers `y,x` must both be `resize!`-able since the number of evaluation points may vary
+between calls to `f!`.
+"""
+struct BatchIntegrand{F,Y,X}
```

Comment:
Actually might be better to declare this as
```
struct BatchIntegrand{Y,X,F}
```
and then make the constructors `BatchIntegrand(f)`, `BatchIntegrand{Y}(f)` and `BatchIntegrand{Y,X}(f)` since these are more idiomatic ways to specify types of containers in Julia.